dist/api_data.py

```diff
@@ -728,6 +728,10 @@ def __ge__(self, other):
             interval in seconds at which to check for files that are inactive and close them''',
             min=1, max=100000),
         ]),
+    Config('generation_drain_timeout_ms', '240000', r'''
```

Comment:
What is the relation to the generation here - is the timeout per generation?

Reply:
I think you should read: "How long can a session wait for a resource to drain".

In the code we can see that `__wt_gen_drain` is called by `__wt_gen_next_drain(session, WT_GEN_EVICT)` which is only called in `__wt_sync_file`.

So this is time maximum time a session would wait for the eviction to drain before timing out. A session should not wait for multiple resources at the same time. However, multiple sessions can wait for the same resource to drain.

@agorrod, can you review my answer?

Reply:
That's right. It's a good observation that this configuration option isn't obvious from a user point of view. I'm OK with it in this case - it's an option we expect to be used primarily by MongoDB in some specific (slow) testing environments.
I'd rather not go down the rabbit hole of generalizing this at the moment.